### DISCUSSION TITLE
Switching wallet whilst connected to a dApp no longer works

### DISCUSSION THREAD

**Discussion by joefitter** [phantom/docs#78] [Help & Support] [answered]
### Summary

When connected to a dApp with Phantom (23.11.0) (Solana) via Solana-wallet-adapter (latest version) nothing happens when I switch wallets.

When I check `window.phantom.solana.publicKey` in the console this still points to the wallet I was using when I connected to the app, and not the new wallet.

Further to this it appears no event is fired on switching wallets - `on("accountChanged")` now does nothing.

This is a critical bug for us as our dApp uses multi-sig transactions and there is no longer any way for the user to switch wallets to sign from multiple.

This bug can be seen across the Solana ecosystem with every app that uses the wallet adapter - eg in Jup.ag you can no longer switch wallets without disconnecting and reconnecting, or refreshing the page

### Example

jup.ag

### Steps to Reproduce

connect to site with phantom wallet a, switch wallet in phantom extension to wallet b - nothing happens.

in browser console check the output of `window.solana.publicKey.toBase58()` - this is still the initial wallet.

### Phantom Version

23.11.0

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hey @joefitter,

Thank you for writing in!

I'm not able to reproduce this on 23.11.0. I'm on Chrome on Mac OS. On jup.ag when I switch to another account in my wallet, I'm either disconnected or switch, if the wallet is previously connected.

Have you tried normal troubleshooting things, like restarting the browser, flushing cache, etc? Do you have anyone else that can reproduce the same behavior?

**Reply by joefitter** [reply]
Thank you for checking this out for me - I can confirm it's still an issue for me using 23.11.0 and Chrome version 116.0.5845.140 (Mac) - there is no way for me to change wallet without first disconnecting and reconnecting.

I have tried on a different mac and it works as expected so I am thinking it could be to do with having many solana wallet apps installed - I'll keep trying and let you know if I can pin down the exact issue.

Also to confirm it's not just me who experiences this, we've had a number of users report the same issue with wallet changes not being detected

**Reply by adamdelphantom** [reply]
Thanks for responding @joefitter! 

We've had some other reports as well since my last response. It appears that it may be isolated to instances in where Solana-Only mode is enabled. Please confirm if this aligns with your experience. Or, if it persists without Solana only mode enabled, please let me know as well.

**Reply by joefitter** [reply]
Apologies for the delay - yes I can confirm it works as expected in multi-chain mode, the bug seems to be isolated to only solana mode

**Comment by adamdelphantom** [comment, marked as answer]
Hey @joefitter, this is a bug. It was also identified here: https://github.com/orgs/phantom/discussions/104

We have a fix coming out in the next few weeks.

**Comment by joefitter** [comment]
Hi @adamdelphantom - thanks for getting back to me.

Just to add some more information - if I connect with Phantom in multi-chain mode it works fine, but only with wallets created using Phantom. If I use a wallet imported via private key there is no way to trigger an account change without physically disconnecting and reconnecting. I assume this is the same root cause as imported wallets are solana only

Thanks, and look forward to the fix 👍

**Comment by joefitter** [comment]
Phantom chrome extension: v23.19.0
Chrome (OSX) 120.0.6099.109 (Official Build) (arm64)

Hi, this was an issue that was resolved a couple of months back but seems to have crept back in recently.

When switching between Solana wallets on a connected dApp (using Solana wallet adapter), nothing happens and the old wallet stays connected.

Expected behaviour: the wallet change is picked up, so the user doesn't need to hard disconnect or refresh the page.

This is especially critical for our multi-sig locking feature in biblio.tech as users need to sign with 2 wallets, the app polls awaiting a wallet change which never fires.

Repro steps:

visit jup.ag
connect wallet
switch wallet in phantom extenstion
observe no change is picked up in the app.

**Reply by adamdelphantom** [reply]
Hi @joefitter,

I'm not able to reproduce this on Jupiter or on our sandbox: https://phantom-sandboxes.vercel.app/sol-adapter-sandbox

Are you still seeing this?
